dashboard/app/controllers/openai_chat_controller.rb

```diff
@@ -72,14 +72,27 @@ def add_content_to_system_prompt(system_prompt, level_instructions, test_file_co
   end
 
   private def read_file_from_s3(key_path)
-    if [:development, :test].include?(rack_env)
+    cache_key = "s3_file_#{key_path}"
```

Comment:
since we don't know how else the shared cache is being used, I would suggest including the bucket name in the cache key.